shm/src/Ship.cpp

```diff
@@ -0,0 +1,36 @@
+#include "../inc/Ship.hpp"
+#include <iostream>
+
+Ship::Ship(size_t id, const std::string& name, size_t speed, size_t maxCrew, size_t capacity)
+    : id_(id)
+    , name_(name)
+    , speed_(speed)
+    , maxCrew_(maxCrew)
+    , capacity_(capacity)
+{}
+
+Ship::Ship(size_t id, size_t speed, size_t maxCrew) 
+    : Ship(id, "Ship", speed, maxCrew, 100)
+{}
+
+Ship& Ship::operator+=(const size_t value) {
+    if ((value + crew_) <= maxCrew_) {
```

Comment:
Redundant inside brackets & crew instead of value would be a better variable name:

(crew + crew_ <= maxCrew_) is enough